common/src/main/java/io/netty/util/ResourceLeakDetector.java

```diff
@@ -407,9 +408,14 @@ public String toString() {
 
             final Object[] array;
             final int removedRecords;
-            synchronized (lastRecords) {
-                array = lastRecords.toArray();
-                removedRecords = this.removedRecords;
+            if (lastRecords != null) {
+                synchronized (lastRecords) {
+                    array = lastRecords.toArray();
+                    removedRecords = this.removedRecords;
+                }
+            } else {
+                removedRecords = 0;
+                array = new Object[0];
```

Comment:
@carl-mastrangelo use `EmptyArrays.EMPTY_OBJECTS` to reduce allocations ;)

Reply:
Done.